client/components/domains/register-domain-step/index.jsx

```diff
@@ -577,7 +578,19 @@ const RegisterDomainStep = React.createClass( {
 				break;
 
 			case 'mappable_but_blacklisted_domain':
-				message = this.translate( 'Domain cannot be mapped to a WordPress.com blog because of blacklisted term.' );
+				if ( domain.toLowerCase().indexOf( 'wordpress' ) > -1 ) {
+					message = this.translate(
+						'Due to {{a1}}trademark policy{{/a1}}, we are not able to allow domains containing <b>WordPress</b> to be registered or mapped here. Please {{a2}}contact support{{/a2}} if you have any questions.',
```

Comment:
You have to "escape" `<b>` the same way you are doing for the links :) Oh, and I think `<strong>` is preferred instead of `<b>`.